vspc/server.py

```diff
@@ -202,20 +207,26 @@ async def handle_telnet(self, reader, writer):
         socket = writer.get_extra_info('socket')
         peer = socket.getpeername()
         LOG.info("%s connected", peer)
-        data = await telnet.read_some()
+        async with timeout(10):
+            data = await telnet.read_some()
         uuid = self.sock_to_uuid.get(socket)
         if uuid is None:
             LOG.error("%s didn't present UUID", peer)
+            await writer.drain()
             writer.close()
+            await writer.wait_closed()
             return
         try:
             while data:
                 await self.save_to_log(uuid, data)
-                data = await telnet.read_some()
+                async with timeout(10):
+                    data = await telnet.read_some()
         finally:
             self.sock_to_uuid.pop(socket, None)
-        LOG.info("%s disconnected", peer)
-        writer.close()
+            LOG.info("%s disconnected", peer)
+            await writer.drain()
```

Comment:
What happens if the other side of our socket is gone? Where does it drain to?

Reply:
We'll have `ConnectionError`... will make the check here.